### PR TITLE
More pathlib fun

### LINKED ISSUE #73: Local file casting issues
Howdy

Trying to cast this media file on my CC Ultra:
```youtube-dl -f "best[width <=? 3840][height <=? 2160]" https://vimeo.com/225888984```
as a local file, results in this traceback:
```
Traceback (most recent call last):
  File "/home/benno/projects/chromecast/catt/catt/http_server.py", line 40, in do_GET
    self.wfile.write(data)
  File "/usr/lib/python3.6/socketserver.py", line 775, in write
    self._sock.sendall(b)
BrokenPipeError: [Errno 32] Broken pipe
192.168.0.188 - - [27/Feb/2018 12:43:54] "GET / HTTP/1.1" 200 -
Traceback (most recent call last):
  File "/home/benno/projects/chromecast/catt/catt/http_server.py", line 40, in do_GET
    self.wfile.write(data)
  File "/usr/lib/python3.6/socketserver.py", line 775, in write
    self._sock.sendall(b)
ConnectionResetError: [Errno 104] Connection reset by peer
```
Served from a DLNA-server, the file plays fine (I did check transcoding settings).

It would be nice to have this fixed, before we axe py2 support. I tried tinkering a bit with the server module, but to no avail.

Upon further investigation, this does not appear to be bitrate related. The Chromecast and the server script are somehow confusing each other when trying to cast certain types of videos. Please note that all the media files that I have had issues with originated from Vimeo (and that regular casting of Vimeo videos with ```catt``` is also flakey).

### PR DESCRIPTION
**Pull request by theychx**
Howdy
Forgot about ```stream_info``` and ```http_server``` also using ```os```. I've included the local file casting fix (removal of "Accept-Ranges" header).

Looking forward (short-term), I'm honestly more keen on working on the save/restore stuff right now. So if it's ok with you, I'd be cool if you could make a ```0.6.0``` release after this is merged.

Fixes #73

### CODE REVIEW

File: catt/http_server.py [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
                         "%a %d %b %Y %H:%M:%S GMT",
-                        time.localtime(os.path.getmtime(filename))
+                        time.localtime(mtime)
                     )
                 )
                 self.end_headers()
 
                 while True:
-                    data = file.read(100 * 1024)
+                    data = mediafile.read(100 * 1024)
```
Comment (theychx) [inline catt/http_server.py]: It does not appear to be possible to do this with ```pathlib```.

Comment (skorokithakis) [reply catt/http_server.py]: What isn't possible, you mean?

**Comment by skorokithakis** [comment]
Looks good to me! I can release after merge, yes, but I'm in the middle of something now, so do remind me if I forget.

File: catt/http_server.py (hunk 1 above)
Comment (theychx) [reply catt/http_server.py]: Partial reading, which is why I used built-in ```open()```.

Comment (skorokithakis) [reply catt/http_server.py]: Ah, yes, unfortunately the `read()` parameter doesn't take ranges.

**Pull request merged**